code/datums/mind.dm

```diff
@@ -86,10 +87,6 @@
 	if(new_character.mind)								//disassociate any mind currently in our new body's mind variable
 		new_character.mind.current = null
 
-	if(istype(current) && islist(current.antag_datums)) //wow apparently current isn't always living good fucking job SOMEONE
-		for(var/i in current.antag_datums)
-			var/datum/antagonist/D = i
-			D.transfer_to_new_body(new_character)

```

Comment:
we need to call remove_innate_effects() on the old body and apply_innate_effects() on the new body for each antag datum.